includes/class-admin-plugins-screen.php

```diff
@@ -0,0 +1,234 @@
+			}
+
+			$plugins[ $slug ] = wp_parse_args( $plugin_info, $default_info );
+		}
+
+		$plugins = $this->order_newspack_plugins_first( $plugins );
+
+		return $plugins;
+	}
+
+	/**
+	 * Set up a custom ordering for the screen, with managed plugins at the top.
+	 *
+	 * @param  array $plugins Array of plugin info.
+	 * @return array Modified $plugins.
+	 */
+	protected function order_newspack_plugins_first( $plugins ) {
+		global $orderby, $status;
+
+		if ( 'all' !== $status ) {
+			return $plugins;
+		}
+
+		$orderby = 'NewspackOrderIndex'; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.OverrideProhibited
+
+		$managed_plugins = Plugin_Manager::get_managed_plugins();
+
+		foreach ( $plugins as $index => $plugin ) {
+			$slug = strpos( $index, '.' ) ? dirname( $index ) : $index;
+
+			if ( isset( $managed_plugins[ $slug ] ) ) {
+				$plugins[ $index ]['NewspackOrderIndex'] = '0-Newspack-Plugin-' . $plugin['Name'];
+				$plugins[ $index ]['Name']               = '&mdash; ' . $plugin['Name'];
+			} elseif ( 'newspack-plugin' === $slug ) {
+				$plugins[ $index ]['NewspackOrderIndex'] = '0-Newspack-Plugin';
+			} else {
+				$plugins[ $index ]['NewspackOrderIndex'] = '1-' . $plugin['Name'];
+			}
+		}
+
+		return $plugins;
+	}
+
+	/**
+	 * Remove 'Activate' and 'Delete' links for uninstalled plugins, and add an 'Install' link.
+	 *
+	 * @param  array  $actions Array of plugin action links.
+	 * @param  string $plugin_file The plugin file.
+	 * @param  array  $plugin_data Information about the plugin.
+	 * @return array  Modified $actions.
+	 */
+	public function modify_action_links( $actions, $plugin_file, $plugin_data ) {
+		$plugin_slug       = isset( $plugin_data['slug'] ) ? $plugin_data['slug'] : $plugin_file;
+		$installed_plugins = Plugin_Manager::get_installed_plugins();
+		$managed_plugins   = array_keys( Plugin_Manager::get_managed_plugins() );
+
+		if ( in_array( $plugin_slug, $managed_plugins ) && ! isset( $installed_plugins[ $plugin_slug ] ) ) {
+			unset( $actions['activate'] );
+			unset( $actions['delete'] );
+			if ( current_user_can( 'install_plugins' ) ) {
+				$actions['install'] = '<a href="' .
+					wp_nonce_url( 'plugins.php?action=newspack_install_plugin&plugin=' . urlencode( $plugin_slug ), 'newspack-install-plugin_' . $plugin_slug, 'install_nonce' ) .
+					'" class="edit" aria-label="' .
+					/* translators: %s - plugin name */
+					esc_attr( sprintf( __( 'Install %s', 'newspack' ), $plugin_data['Name'] ) ) .
+					'">' .
+					__( 'Install', 'newspack' ) .
+					'</a>';
+			}
+		}
+
+		return $actions;
+	}
+
+	/**
+	 * Install a plugin when the installation link is clicked.
+	 */
+	public function handle_plugin_install() {
+		if ( ! current_user_can( 'install_plugins' ) ) {
+			wp_die( esc_html__( 'Sorry, you are not allowed to install plugins.', 'newspack' ) );
+		}
+
+		$plugin = isset( $_REQUEST['plugin'] ) ? sanitize_title( $_REQUEST['plugin'] ) : ''; // phpcs:ignore WordPress.VIP.SuperGlobalInputUsage
```

Comment:
Last note: how about replacing this logic with `filter_input` like the other case, which would allow removal of the PHPCS ignore rule?

`$plugin = filter_input( INPUT_GET, 'plugin', FILTER_SANITIZE_STRING );`

Reply:
Resolved in https://github.com/Automattic/newspack-plugin/pull/13/commits/fa1279038288db47cadc17e39e0a8c99bef61f6b